### DISCUSSION TITLE
styles -> options requires all style options to be defined even though they are merged, is this intentional?

### DISCUSSION THREAD

**Discussion by thenick775** [gilbarbara/react-joyride#968] [Q&A] [answered]
Writing a component as follows:

```
<Joyride
  ...
  styles={{
    options: {
      zIndex: zIndex
    }
  }}
  ...
/>
```

with:
```
"react": "^18.2.0",
"react-joyride": "^2.6.2",
"typescript": "^5.1.3",
```

Throws an error with typescript of the form:
```
Type '{ zIndex: number; }' is missing the following properties from type 'StylesOptions': arrowColor, backgroundColor, beaconSize, overlayColor, and 3 more.
```

Given the interface definition for options:
```
interface StylesOptions {
    arrowColor: string;
    backgroundColor: string;
    beaconSize: number;
    overlayColor: string;
    primaryColor: string;
    spotlightShadow: string;
    textColor: string;
    width?: string | number;
    zIndex: number;
}
```

Is this intentional given the example for styling is also missing types from the current definition, and that `styles.js` performs a deep merge on the options we provide (merging with the default options), as well as having styles defined as `styles?: PartialDeep<Styles>;`?

Thank you in advance for any insights!

**Comment by JandenMa** [comment]
I got the same issue when upgrading to v2.6!
@gilbarbara I read your source codes today and noticed that when you migrated source codes from JS to TS, in the `StyleOptions` type definition, most of the options (except `width`) are mandatory. IMHO I don't think it is correct. Bcoz you have used `deepmerge` to combine the `defaultOptions` and the customized options, that means I only need to pass some of the options that I need to override, not all of them.
Probably I could be wrong. 
However, as a user, it seems not friendly to me.
Looking forward to your reply. Thank you!

**Reply by JandenMa** [reply]
now I have to downgrade to v2.5.5

**Comment by gilbarbara** [comment, marked as answer]
Fixed in v2.7.0

**Reply by JandenMa** [reply]
Thank you! I'll check it.

**Reply by thenick775** [reply]
Thank you, confirmed as a fix.

Just want to say, thanks for all you do as a maintainer! I appreciate your dedication, insights, and the value your packages provide!

All the best,
Nick

**Reply by gilbarbara** [reply]
> Just want to say, thanks for all you do as a maintainer! I appreciate your dedication, insights, and the value your packages provide!

Aww, thanks! Paying it forward to the community!

**Reply by JandenMa** [reply]
Tested it, works well! Thank you!
